Monika After Story/game/zz_backgrounds.rpy

```diff
@@ -0,0 +1,518 @@
+
+            IN:
+                background_id:
+                    id that defines the background object
+                    NOTE: Must be unique
+
+                prompt:
+                    button label for this bg
+
+                image_day:
+                    the renpy.image object we use for this bg during the day
+                    NOTE: Mandatory
+
+                image_night:
+                    the renpy.image object we use for this bg during the night
+                    NOTE: Mandatory
+
+                image_rain_day:
+                    the image tag we use for the background while it's raining (day)
+                    (Default: None, not required)
+
+                image_rain_night:
+                    the image tag we use for the background while it's raining (night)
+                    (Default: None, not required)
+
+                image_overcast_day:
+                    the image tag we use for the background while it's overcast (day)
+                    (Default: None, not required)
+
+                image_overcast_night:
+                    the image tag we use for the background while it's overcast (night)
+                    (Default: None, not required)
+
+                image_snow_day:
+                    the image tag we use for the background while it's snowing (day)
+                    (Default: None, not required)
+
+                image_snow_night:
+                    the image tag we use for the background while it's snowing (night)
+                    (Default: None, not required)
+
+                hide_calendar:
+                    whether or not we want to display the calendar
+                    (Default: False)
+
+                hide_masks:
+                    weather or not we want to show the windows
+                    (Default: False)
+
+                disable_progressive:
+                    weather or not we want to disable progressive weather
+                    (Default: None, if hide masks is true and this is not provided, we assume True, otherwise False)
+
+                unlocked:
+                    whether or not this background starts unlocked
+                    (Default: False)
+
+                entry_pp:
+                    Entry programming point for the background
+                    (Default: None)
+
+                exit_pp:
+                    Exit programming point for this background
+                    (Default: None)
+            """
+
+            if background_id in self.mas_background.BACKGROUND_MAP:
+                raise Exception("duplicate background ID")
+
+            self.background_id = background_id
+            self.prompt = prompt
+            self.image_day = image_day
+            self.image_night = image_night
+
+            #Handle rain BGs
+            if image_rain_day:
+                self.image_rain_day = image_rain_day
+            else:
+                self.image_rain_day = image_day
```

Comment:
since you repeat this pattern of if statements, i would make a private function like:
```python
def __handleBG(self, desired_bg, fallback_bg):
    if desired_bg:
        return desired_bg
    return fallback_bg
```
then you can use that here like:
```python
self.image_rain_day = self.__handle_bg(image_rain_day, image_day)
```